Challenge/javascript-onboarding/src/problem2.js

```diff
@@ -1,6 +1,49 @@
+function isDuplicate(string, index) {
+  return (
+    string[index - 1] === string[index] || string[index] === string[index + 1]
+  );
+}
+
+function deleteDuplicateCharacter(string) {
+  return Array.from(string)
+    .filter((_, index) => !isDuplicate(string, index))
+    .join("");
+}
+
+function deleteAllDuplicateCharacter(string) {
+  const result = deleteDuplicateCharacter(string);
+
+  return string.length !== result.length
+    ? deleteAllDuplicateCharacter(result)
+    : result;
```

Comment:
우테코라면 삼항 연산자를 별로 안 좋아해서 안 쓸 것 같아요

```suggestion
if(string.length===result.length){
    return string;
}
return deleteAllDuplicateCharacter(result);
```

Reply:
지금 같은 경우는 삼항 연산자가 길어져서 `if`문을 사용한 방식이 조금 더 읽기 쉬워 보이네요

그러면 혹시 아래와 같이 한 줄로 표현 가능한 경우 삼항 연산자 사용은 어떻게 생각하시나요?

```javascript
function something() {
  const result = "Result";
  return result ? "Success" : "Fail";
}
```